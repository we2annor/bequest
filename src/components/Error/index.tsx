import React from "react";

interface ErrorProps {
  message: string;
}

const Error: React.FC<ErrorProps> = ({ message }) => {
  return (
    <div data-testid='error' className='error'>
      {message}
    </div>
  );
};

export default Error;
