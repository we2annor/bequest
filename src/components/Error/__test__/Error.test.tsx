import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import Error from "../index";
import renderer from "react-test-renderer";

afterEach(cleanup);

describe("On Mount", () => {
  test("Renders without crushing", () => {
    const div = document.createElement("div");
    const message = "";
    ReactDOM.render(<Error message={message}></Error>, div);
  });

  test("Renders correctly", () => {
    const message = "Error occured";
    const { getByTestId } = render(<Error message={message} />);
    expect(getByTestId("error").textContent).toBe("Error occured");
  });

  test("Render correctly with a different message", () => {
    const message = "Failed to load the api";
    const { getByTestId } = render(<Error message={message} />);
    expect(getByTestId("error").textContent).toBe("Failed to load the api");
  });

  test("Matches a snapshot", () => {
    const tree = renderer.create(<Error message={"Error occured"} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Maches another snapshot", () => {
    const tree = renderer
      .create(<Error message='Failed to load the api' />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
