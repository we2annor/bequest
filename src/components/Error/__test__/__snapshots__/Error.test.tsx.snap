// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`On Mount Maches another snapshot 1`] = `
<div
  className="error"
  data-testid="error"
>
  Failed to load the api
</div>
`;

exports[`On Mount Matches a snapshot 1`] = `
<div
  className="error"
  data-testid="error"
>
  Error occured
</div>
`;
